### PR TITLE
improve(multicallerclient): Log multicaller execution failures

### PR DESCRIPTION
**Pull request by nicholaspai**
This  PR tries to filter out any errors that occur when a transaction is successfully simulated but then fails when it is mined. This can happen for the relayer commonly if they get front-run by another 
relay, which is expected behavior

### CODE REVIEW

File: src/clients/MultiCallerClient.ts [hunk 1]
```diff
@@ -117,26 +117,29 @@ export class MultiCallerClient {
...
+          return [chainId, { result: result.reason, isError: !knownRevertReasons.has(result.reason) }];
         }
       })
     );
 
     // We need to iterate over the results to determine if any of the transactions failed.
     // If any of the transactions failed, we need to log the results and throw an error. However, we want to
     // only log the results once, so we need to collate the results into a single object.
-    const failedChains = Object.entries(txnHashes)
-      .filter(([, { isError }]) => isError)
```
Comment (james-a-morris) [inline src/clients/MultiCallerClient.ts:129]: Could we improve this filter and remove the ignorable reversions here?

**Review by james-a-morris — COMMENTED** [review]
Left a comment

File: src/clients/MultiCallerClient.ts (hunk 1 above)
Comment (nicholaspai) [reply src/clients/MultiCallerClient.ts:129]: Actually I think this filter won't help that much, first things first i'd like to just log the failures better

File: src/clients/MultiCallerClient.ts [hunk 2]
```diff
@@ -99,7 +99,7 @@ export class MultiCallerClient {
     const chainIds = [...new Set(Object.keys(this.valueTxns).concat(Object.keys(this.txns)))];
 
     // One promise per chain for parallel execution.
-    const results = await Promise.allSettled(
+    const resultsByChain = await Promise.allSettled(
```
Comment (nicholaspai) [inline src/clients/MultiCallerClient.ts:102]: This is a more intuitive name IMO as results is actually a list of PromiseSettledResults for each chain. So i'm curious what a failure looks like if one of the txns fails

**Pull request merged**